### PR TITLE
JIT: fix some issues in fgIsThrowHlpBlk

### LINKED ISSUE #58372: JIT: [QualityWeek] Random edge counts assert in System.Text.Encoding.Tests
Random edge counts triggered an assert running one of the libraries tests on x64 windows.

https://dev.azure.com/dnceng/public/_build/results?buildId=1325466&view=ms.vss-test-web.build-test-results-tab

```
C:\h\w\B80B09E6\p\dotnet.exe" exec --runtimeconfig System.Text.Encoding.Tests.runtimeconfig.json --depsfile System.Text.Encoding.Tests.deps.json xunit.console.dll System.Text.Encoding.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing  

COMPlus_JitRandomEdgeCounts=1
COMPlus_JitRandomGuardedDevirtualization=1
COMPlus_ReadyToRun=0
COMPlus_TC_QuickJitForLoops=1
COMPlus_TieredCompilation=1
COMPlus_TieredPGO=1

Assert failure(PID 9328 [0x00002470], Thread: 8064 [0x1f80]): Assertion failed '*node == call' in 'System.Text.Encoding:GetBytes(System.String):System.Byte[]:this' during 'Generate code' (IL size 62)
```

### PR DESCRIPTION
**Pull request by AndyAyersMS**
We may see no return calls that are not the roots of their trees, so don't
assert that the last node must be a call.

Update the list of helpers we check for to reflect current practice.

Remove the unused SCK_PAUSE_EXEC.

Fixes #58372.

### CODE REVIEW

File: src/coreclr/jit/compiler.hpp [hunk 1]
```diff
@@ -2612,32 +2612,19 @@ inline bool Compiler::fgIsThrowHlpBlk(BasicBlock* block)
...
 
-    if (!call || (call->gtOper != GT_CALL))
+    if ((call == nullptr) || (call->gtOper != GT_CALL))
     {
         return false;
     }
 
     if (!((call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_RNGCHKFAIL)) ||
           (call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_THROWDIVZERO)) ||
-          (call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_THROWNULLREF)) ||
```
Comment (EgorBo) [inline src/coreclr/jit/compiler.hpp:2640]: I guess it should also be removed from comments for `fgIsThrowHlpBlk` function.

Also looks like it's unused in the whole repo and can be removed?

Comment (EgorBo) [reply src/coreclr/jit/compiler.hpp:2640]: At some point I wanted to use it for poor-man's "meaningful" nullreference exceptions via expanding implicit nullchecks to flow-level checks 🙂

Comment (AndyAyersMS) [reply src/coreclr/jit/compiler.hpp:2640]: I don't know the back story of `HELP_THROWNULLREF` -- @erozenfeld do you recall if this was something we added just for LLILC (per https://github.com/dotnet/coreclr/pull/1501)?

Comment (erozenfeld) [reply src/coreclr/jit/compiler.hpp:2640]: Yes, I think so.

**Review by EgorBo — APPROVED** [review]

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**